ingestor/adx/uploader.go

```diff
@@ -388,3 +390,32 @@ type clusterDetails struct {
 	ClockDescription       string    `kusto:"ClockDescription"`
 	RuntimeDescription     string    `kusto:"RuntimeDescription"`
 }
+
+// countingReader wraps an io.Reader and counts bytes read without allocations.
+type countingReader struct {
```

Comment:
Can we move this to something like `pkg/reader`.  Looks general enought to be useful in other places.